subprojects/docs/src/docs/userguide/extending-gradle/task_configuration_avoidance.adoc

```diff
@@ -57,6 +57,25 @@ If you are configuring a task by name, you will need to use the configuration av
 
 In the event you still need to get access to a Task instance, you can use link:{javadocPath}/org/gradle/api/tasks/TaskCollection.html#named-java.lang.String-[TaskCollection.named(java.lang.String)] and call link:{javadocPath}/org/gradle/api/provider/Provider.html#get--[Provider.get()].  This will cause the task to be created/configured, but everything should work as it has with the eager APIs.
 
+[[sec:how_do_i_order_tasks]]
+=== How do I order tasks?
+
+In case, when task relationships needs to be established (`dependsOn`, `finalizedBy`, `mustRunAfter`, `shouldRunAfter`), configuration avoidance can also help to avoid configuring tasks.
```

Comment:
I think I will yield on the "In case, when" to simply "when" case.
Apart from this, you might misunderstand this first sentence - this is an introduction to a section, and needs to give some context to the reader.

Though the ending might need some work:
>  configuration avoidance can also help to avoid configuring tasks

is pretty redundant, I might cook up something better.